src/Database/Query.php

```diff
@@ -1393,6 +1394,27 @@ public function into($table)
     }
 
     /**
+     * Creates an expression that refers to an identifier. Identifiers are used to refer to field names and allow
+     * the SQL compiler to apply quotes or escape the identifier.
+     *
+     * The value is used as is, and you might be required to use aliases or include the table reference in
+     * the identifier. Do not use this method to inject SQL methods or logical statements.
+     *
+     * ### Example
+     *
+     * ```
+     * $query->newExp()->lte('count', $query->identifier('total'));
+     * ```
+     *
+     * @param string $identifier The identifier for an expression
+     * @return ExpressionInterface
+     */
+    public function identifier($identifier)
+    {
+        return new IdentifierExpression($identifier);
```

Comment:
@thinkingmedia There is no public, non-internal API that uses `IdentifierExpression`, currently the only place where the core tests for / expects `IdentifierExpression`, is in `IdentifierQuoter::quoteExpression()`, `IdentifierQuoter::_quoteIdentifierExpression()`, and `QueryExpression::_calculateType()`.

My concern was just that the API doesn't provide a realiable contract, which I personally like too avoid whenever possible. With respect to the mentioned methods, if I cannot be sure that the value returned is an actual `IdentifierExpression` object, then I cannot be sure that my identifiers will actually get quoted, that the column type for the identifiers can be determined, and that things don't error out, as the public API does not guarantee the required expression implementation.

I know, I'm nitpicking, I'd write tests anyways to ensure things work as expected, etc. I'm just not a big fan of such situations, and there already are some places in the core where contract violations slipped in because of the usage of insufficient interfaces, so I thought I'd mention it.